src/distributing-policies/secure-supply-chain.md

```diff
@@ -0,0 +1,387 @@
+
+To sign a policy users can execute the `cosign sign` command passing the `--key`
+command line argument with their private key file:
+
+```bash
+cosign sign --key cosign.key ghcr.io/jvanz/policies/user-group-psp:latest
+an error occurred: no provider found for that key reference, will try to load key from disk...
+Enter password for private key:
+Pushing signature to: ghcr.io/jvanz/policies/user-group-psp
+```
+
+This command will sign the policy and publish the signature data in the registry
+as well. Now the policies are ready to be used in a Kubewarden installation with signature
+verification enabled.
+
+
+### Keyless signing
+
+Many times the policies are automatically built in CI/CD pipelines which complicates
+the key generation process. For these situations and others where key generation is not
+possible, `cosign` has an experimental feature, keyless signing. This allows the users to
+sign policies without a previously created key. By using
+[Fulcio](https://github.com/sigstore/fulcio), ephemeral keys are generated that can be used to
+sign the artifact only once.
+
+To ensure the identity of the user for these keys, Fulcio uses an OIDC provider
+to generate the ephemeral keys. This means that, during the signing process users
+must sign-in using one of the OIDC providers, as shown in the below image:
+
+![Sigstore OIDC](./sigstore_oidc.png)
+
+
+After signing in, `cosign` will finish the signing process.
+
+However, as of this writing, this feature is not enabled by default
+in `cosign`. So, it's necessary to enable this experimental feature before
+proceeding with keyless signing as shown below:
+
+```bash
+ COSIGN_EXPERIMENTAL=1 cosign sign ghcr.io/jvanz/policies/user-group-psp:latest
+Generating ephemeral keys...
+Retrieving signed certificate...
+Your browser will now be opened to:
+https://oauth2.sigstore.dev/auth/auth?access_type=online&client_id=sigstore&code_challenge=<REDACTED>&code_challenge_method=S256&nonce=<REDACTED>&redirect_uri=http%3A%2F%2Flocalhost%3A34021%2Fauth%2Fcallback&response_type=code&scope=openid+email&state=<REDACTED>
+client.go:196: root pinning is not supported in Spec 1.0.19
+Successfully verified SCT...
+tlog entry created with index: 1819248
+Pushing signature to: ghcr.io/jvanz/policies/user-group-psp
+```
+
+### How to sign artefacts in Github workflows
+
+`cosign` does not require the user to log on to an OIDC provider when using keyless signing
+in the context of a GitHub action environment. In this scenario,
+the Github token available during the execution of the workflow will be used to
+authenticate the user and generate the ephemeral keys.
+
+## Verifying policies
+
+Users can check if a policy is properly signed with  `cosign` or `kwctl`.
+Both have similar command line flags which allow users to check the policy
+signatures. To check if the binary is signed with a key, the users can use `kwctl`
+like this:
```

Comment:
One can list all signatures with `kwctl inspect`, and verify them with `kwctl verify`.